app/database_etl/location_utils/location_util_functions.py

```diff
@@ -124,29 +124,29 @@ def row_in_feature_layer(row: pd.Series, feature_layer: FeatureLayer) -> bool:
 # Get pin latitude and longitude for a record
 # Param geo_dfs = dictionary of city, state, and country DFs
 def get_record_coordinates(record: pd.Series, geo_dfs: dict) -> Tuple:
-    pin_regions = [record['country']]
-    pin_region_type = 'country' if record['country'] is not None else ''
+    pin_lat = None
+    pin_lng = None
 
-    for region_type in ['state', 'city']:
+    for region_type in ['country', 'state', 'city']:
         if record[region_type] and len(record[region_type]) > 0:
             pin_regions = record[region_type]
-            pin_region_type = region_type
+            geo_df = geo_dfs[region_type]
+            col_name = f"{region_type}_name"
+            pin_lats = [geo_df[geo_df[col_name] == region_name].iloc[0]['latitude'] for region_name
+                        in pin_regions if pd.notnull(geo_df[geo_df[col_name] == region_name].iloc[0]['latitude'])]
+            pin_lngs = [geo_df[geo_df[col_name] == region_name].iloc[0]['longitude'] for region_name
+                        in pin_regions if pd.notnull(geo_df[geo_df[col_name] == region_name].iloc[0]['longitude'])]
+            if len(pin_lats) > 0:
```

Comment:
There is an edge case here where pin_lats and pin_longs don't have the same length that it might be worth addressing. The code here assumes that latitude is NA iff longitude is NA

Reply:
in theory yes, but in practice do you actually think there is a case where there is a lat and not a lng? I don't know about that

Reply:
i guess you could just add and len(pin_lngs) > 0

Reply:
This should never happen. The api call will return both a lat and lng. I can add this extra check for good measure but confident that wouldn't happen